src/shogun/base/SGObject.h

```diff
@@ -963,6 +991,14 @@ class CSGObject
 	 */
 	AnyParameter get_parameter(const BaseTag& _tag) const;
 
+	/** Getter for a class function, identified by a BaseTag.
+	 * Throws an exception if the class does not have such a parameter.
+	 *
+	 * @param _tag name information of parameter
+	 * @return value of the parameter identified by the input tag
+	 */
+	AnyParameter get_function(const BaseTag& _tag) const;
```

Comment:
this function getter should guarantee that a registered non const class function cannot be run with `get` but instead has to be invoked using `run`